From e64de8b44bb2a39ebfece92cd464237e4eeb9fba Mon Sep 17 00:00:00 2001
From: Jari Aalto <jari.aalto@cante.net>
Date: Wed, 12 Mar 2008 17:02:08 +0200
Subject: [PATCH] Makefile.in: rewrite install target

---
 Makefile.in |   36 +++++++++++++++++++++++++++++++++++-
 1 files changed, 35 insertions(+), 1 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 66daa8a..98f9a04 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -23,6 +23,26 @@ MKDEP	=@MKDEP@
 BINST	=$(INSTALL) -m755
 MINST	=$(INSTALL) -m644
 
+
+PACKAGE		= tirc
+DESTDIR		=
+man_prefix	= $(prefix)/share
+
+BINDIR		= $(DESTDIR)$(exec_prefix)/bin
+SHAREDIR	= $(DESTDIR)$(prefix)/share/$(PACKAGE)
+LIBDIR		= $(DESTDIR)$(prefix)/lib/$(PACKAGE)
+SBINDIR		= $(DESTDIR)$(exec_prefix)/sbin
+ETCDIR		= $(DESTDIR)/etc/$(PACKAGE)
+DOCDIR		= $(DESTDIR)$(prefix)/share/doc
+MANDIR		= $(DESTDIR)$(man_prefix)/man
+MANDIR1		= $(MANDIR)/man1
+
+INSTALL_BIN	= $(INSTALL) -m 755
+INSTALL_DATA	= $(INSTALL) -m 644
+
+INSTALL_OBJS_BIN   = $(PACKAGE)
+INSTALL_OBJS_MAN1  = *.1
+
 .c.o:
 	$(CC) $(CFLAGS) $(CPPFLAGS) -c $<
 
@@ -47,13 +67,27 @@ clean:
 distclean: clean
 	rm -f config.cache config.status config.log config.h Makefile conftest* ht.h regex.h
 
-install: tirc tirc.1
+install-old: tirc tirc.1
 	$(BINST) tirc $(bindir)
 #	$(BINST) tircbug $(bindir)
 	$(MINST) tirc.1 $(mandir)/man1
 #	rm -f $(mandir)/man1/tircbug.1
 #	ln -s $(mandir)/man1/tirc.1 $(mandir)/man1/tircbug.1
 
+install-man:
+	# install-man
+	$(INSTALL_BIN) -d $(MANDIR1)
+	$(INSTALL_DATA) $(INSTALL_OBJS_MAN1) $(MANDIR1)
+
+install-bin:
+	# install-bin
+	$(INSTALL_BIN) -d $(BINDIR)
+	$(INSTALL_BIN) -s $(INSTALL_OBJS_BIN) $(BINDIR)
+
+install: install-bin install-man
+
+.PHONY: clean distclean realclean install install-bin install-man
+
 tags:	*.c *.h
 	$(CTAGS) *.c *.h
 
-- 
1.5.3.8

